compiler/rustc_metadata/src/native_libs.rs

```diff
@@ -523,8 +523,16 @@ impl<'tcx> Collector<'tcx> {
         let calling_convention = if self.tcx.sess.target.arch == "x86" {
             match abi {
                 Abi::C { .. } | Abi::Cdecl { .. } => DllCallingConvention::C,
-                Abi::Stdcall { .. } | Abi::System { .. } => {
-                    DllCallingConvention::Stdcall(self.i686_arg_list_size(item))
+                Abi::Stdcall { .. } => DllCallingConvention::Stdcall(self.i686_arg_list_size(item)),
+                Abi::System { .. } => {
```

Comment:
Is this even reachable? This looks like it is duplicating the logic in adjust_abi.

Reply:
It is reachable. The ABI being matched on comes from the HIR for the extern block, without going through `adjust_abi`. This match looks similar to the logic in `adjust_abi`, but I don't think it can be cleanly deduplicated because this function will emit errors for some ABIs that `adjust_abi` will turn into `C`.

Reply:
That's unfortunate but I have no idea why this is happening or where these two functions are called. If the logic needs to be duplicated here then please add a reference to the other copy of the same logic.